pkg/specs/stategetter/client.go

```diff
@@ -0,0 +1,62 @@
+package stategetter
+
+import (
+	"context"
+	"encoding/base64"
+	"path/filepath"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/ship/pkg/state"
+	"github.com/spf13/afero"
+)
+
+type StateGetter struct {
+	Logger   log.Logger
+	Contents *state.UpstreamContents
+	Fs       afero.Afero
+}
+
+func NewStateGetter(fs afero.Afero, logger log.Logger, contents *state.UpstreamContents) *StateGetter {
+	return &StateGetter{
+		Contents: contents,
+		Fs:       fs,
+		Logger:   logger,
+	}
+}
+
+func (g *StateGetter) GetFiles(
+	ctx context.Context,
+	upstream string,
+	destinationPath string,
```

Comment:
What are possible values for `destinationPath` here? Is it possible that `destinationPath` might cause us to `Fs.RemoveAll` some important local work that wasn't generated by ship?

Reply:
Destination path is an internal tempdir, so there's nothing sensitive to remove - I'm not quite sure why we're removing it here

I think gogetter requires that the dir not exist, but I don't know why that's been carried over into the github client and stateclient

Reply:
And that removal has been removed

Reply:
Okay cool :shipit: